tests/run/t6637.scala

```diff
@@ -1,8 +0,0 @@
-
-object Test extends dotty.runtime.LegacyApp {
```

Comment:
maybe it would be better to remove the `.isInstanceOf[A with B]` part, while keeping the other parts of the test?